xarray/core/resample.py

```diff
@@ -339,7 +338,16 @@ def map(
         applied = (func(ds, *args, **kwargs) for ds in self._iter_grouped())
         combined = self._combine(applied)
 
-        return combined.rename({self._resample_dim: self._dim})
+        # If the aggregation function didn't drop the original resampling
+        # dimension, then we need to do so before we can rename the proxy
+        # dimension we used.
+        if self._dim in combined.coords:
+            combined = combined.drop_vars(self._dim)
+
+        if RESAMPLE_DIM in combined.dims:
```

Comment:
We'd need cumsum and cumprod tests for resample

Reply:
We technically have with the docstring. Still want one?

Reply:
Its nice during development to just run `test_groupby.py`

Reply:
done